dotCMS/src/integration-test/java/com/dotmarketing/business/web/UserWebAPIImplTest.java

```diff
@@ -0,0 +1,137 @@
+        request.setAttribute(WebKeys.USER, frontEndUser);
+        return request;
+    }
+
+    private HttpServletRequest backEndRequest() {
+        final HttpServletRequest request = anonymousRequest();
+        assertTrue("backEndUser has backend role", backEndUser.isBackendUser());
+        request.setAttribute(WebKeys.USER, backEndUser);
+        return request;
+    }
+
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+
+        frontEndUser = new UserDataGen().roles(APILocator.getRoleAPI().loadFrontEndUserRole()).nextPersisted();
+
+        backEndUser = new UserDataGen().roles(APILocator.getRoleAPI().loadBackEndUserRole()).nextPersisted();
+
+
+        userWebAPI = new UserWebAPIImpl();
+
+    }
+
+
+
+    /**
+     * Validates that only front end users can be returned from the call to
+     * userWebAPI.getLoggedInFrontendUser.
+     */
+    @Test
+    public void test_getLoggedInFrontendUser() {
+
+        assert (userWebAPI.getLoggedInFrontendUser(frontEndRequest()).equals(frontEndUser));
+        assert (userWebAPI.getLoggedInFrontendUser(backEndRequest()) == null);
+
+        assert (userWebAPI.getLoggedInFrontendUser(anonymousRequest()) == null);
+
+    }
+
+    /**
+     * validates that getLoggedInUser returns any user that is logged into the dotCMS front end or back
+     * end
+     * 
+     */
+    @Test
+    public void test_getLoggedInUser() {
+
+        assert (userWebAPI.getLoggedInUser(frontEndRequest()).equals(frontEndUser));
+        assert (userWebAPI.getLoggedInUser(backEndRequest()).equals(backEndUser));
+
+
+        assert (userWebAPI.getLoggedInUser(anonymousRequest()) == null);
+
+    }
+
+    /**
+     * validates that getUser returns any user that is logged into the dotCMS front end or back end OR
+     * returns CMS_ANON if not set
+     * 
+     */
+    @Test
+    public void test_getUser() {
+
+        assert (userWebAPI.getUser(backEndRequest()).equals(backEndUser));
+        assert (userWebAPI.getUser(frontEndRequest()).equals(frontEndUser));
+
+        assert (userWebAPI.getUser(anonymousRequest()).equals(userWebAPI.getAnonymousUserNoThrow()));
+
+    }
+
+    /**
+     * validates that isLoggedToBackend true if front end or back end OR returns CMS_ANON if not set
+     * 
+     */
+    @Test
+    public void test_isLoggedToBackend() {
+
+        assert (userWebAPI.isLoggedToBackend(backEndRequest()) == true);
+        assert (userWebAPI.isLoggedToBackend(frontEndRequest()) == false);
+
+        assert (userWebAPI.isLoggedToBackend(anonymousRequest()) == false);
+
+    }
+
+
+    /**
+     * validates that getUser returns any user that is logged into the dotCMS front end or back end OR
+     * returns CMS_ANON if not set
+     * 
+     */
+    @Test
+    public void test_isLoggedToFrontend() {
+
+        assert (userWebAPI.isLoggedToFrontend(backEndRequest()) == false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary comparisons in boolean expressions](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5106108)